alerts/ui/src/components/filters/FilterPills.jsx

```diff
@@ -3,27 +3,141 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from "react"
+import React, { useEffect } from "react"
 
 import { Pill, Stack } from "@cloudoperators/juno-ui-components"
-import { useActiveFilters, useFilterActions } from "../../hooks/useAppStore"
+import {
+  useActiveFilters,
+  useFilterActions,
+  useFilterPills,
+} from "../../hooks/useAppStore"
 
 const FilterPills = () => {
   const activeFilters = useActiveFilters()
-  const { removeActiveFilter } = useFilterActions()
+  const filters = useFilterPills()
+  const { removeActiveFilter, addActiveFilter, setFilterPills } =
+    useFilterActions()
+  // useEffect Hook zur Aktualisierung der Filter
+  useEffect(() => {
+    const pills = (prevFilters) => {
+      const newFilters = { ...prevFilters }
+
+      for (let key in newFilters) {
+        if (newFilters.hasOwnProperty(key)) {
+          if (activeFilters[key]) {
+            const activeSet = new Set(activeFilters[key])
+
+            newFilters[key] = newFilters[key].map((item) => {
+              if (activeSet.has(item.value)) {
+                return { ...item, active: true }
+              }
+              return item
+            })
+
+            activeFilters[key].forEach((value) => {
+              if (!newFilters[key].some((item) => item.value === value)) {
+                newFilters[key].push({ value, active: true })
+              }
+            })
+          }
+        }
+      }
+
+      // Hinzufügen neuer Kategorien aus activeFilters
+      for (let key in activeFilters) {
+        if (!newFilters.hasOwnProperty(key)) {
+          newFilters[key] = activeFilters[key].map((value) => ({
+            value,
+            active: true,
+          }))
+        }
+      }
+
+      return newFilters
+    }
+    setFilterPills(pills(filters))
+  }, [activeFilters])
+
+  const pauseFilter = (key, value) => {
```

Comment:
Why are this and the following functions that all modify the filters defined here and not as actions in the store?

Reply:
Because they are just important for this component and would lead to a more filled Store. FilterPills are just useful in the store for deeplink Implantation which is not done now.